src/TreeSitter/Node.hs

```diff
@@ -29,6 +30,10 @@ data TSPoint = TSPoint { pointRow :: !Word32, pointColumn :: !Word32 }
 data TSNode = TSNode !Word32 !Word32 !Word32 !Word32 !(Ptr ()) !(Ptr ())
   deriving (Show, Eq, Generic)
 
+newtype FieldId = FieldId { getFieldId :: Word16 }
+  deriving (Eq, Ord, Show, Storable)
```

Comment:
I’m introducing a `newtype` for this as a (small) step towards a higher-level `haskell-tree-sitter` API.

This is added here rather than in `Cursor.hs` since there is additionally `TSNode` API for interacting with field IDs; we don’t currently export that, but we do use this type in `Cursor.hs`.